src/seed.rs

```diff
@@ -62,22 +62,22 @@ pub fn next_double(lo: f64, hi: f64, mut s0: Seed) -> (f64, Seed) {
 }
 
 pub fn mix64(x: u64) -> u64 {
-    let y = (x ^ (x >> 33)) * 0xff51afd7ed558ccd;
-    let z = (y ^ (y >> 33)) * 0xc4ceb9fe1a85ec53;
+    let y = (x ^ (x >> 33)).wrapping_mul(0xff51afd7ed558ccd);
+    let z = (y ^ (y >> 33)).wrapping_mul(0xc4ceb9fe1a85ec53);
     z ^ (z >> 33)
 }
 
 #[allow(overflowing_literals)]
 #[allow(exceeding_bitshifts)]
 pub fn mix32(x: u32) -> u32 {
-    let y = (x ^ (x >> 33)) * 0xff51afd7ed558ccd;
-    let z = (y ^ (y >> 33)) * 0xc4ceb9fe1a85ec53;
+    let y = (x ^ (x >> 33)).wrapping_mul(0xff51afd7ed558ccd);
+    let z = (y ^ (y >> 33)).wrapping_mul(0xc4ceb9fe1a85ec53);
     z >> 32
 }
 
 pub fn mix64_variant13(x: u64) -> u64 {
-    let y = (x ^ (x >> 30)) * 0xbf58476d1ce4e5b9;
-    let z = (y ^ (y >> 27)) * 0x94d049bb133111eb;
+    let y = (x ^ (x >> 30)).wrapping_mul(0xbf58476d1ce4e5b9);
```

Comment:
I had to do these as I was getting panics from rust when running tests that indirectly called `seed` fns.

Reply:
really? what does this do?

Reply:
so rust will panic by default on overflow/underflow from ops, this just goes back to the default you'd expect in other languages